mne/time_frequency/tfr.py

```diff
@@ -470,28 +489,13 @@ def _induced_power(data, sfreq, frequencies, use_fft=True, n_cycles=7,
     # Precompute wavelets for given frequency range to save time
     Ws = morlet(sfreq, frequencies, n_cycles=n_cycles, zero_mean=zero_mean)
 
-    if n_jobs == 1:
-        psd = np.empty((n_channels, n_frequencies, n_times))
-        plf = np.empty((n_channels, n_frequencies, n_times), dtype=np.complex)
-
-        for c in range(n_channels):
-            X = data[:, c, :]
-            this_psd, this_plf = _time_frequency(X, Ws, use_fft)
-            psd[c], plf[c] = this_psd[:, ::decim], this_plf[:, ::decim]
-    else:
-        parallel, my_time_frequency, _ = parallel_func(_time_frequency, n_jobs)
-
-        psd_plf = parallel(my_time_frequency(np.squeeze(data[:, c, :]),
-                                             Ws, use_fft)
-                           for c in range(n_channels))
-
-        psd = np.zeros((n_channels, n_frequencies, n_times))
-        plf = np.zeros((n_channels, n_frequencies, n_times), dtype=np.complex)
-        for c, (psd_c, plf_c) in enumerate(psd_plf):
-            psd[c, :, :], plf[c, :, :] = psd_c[:, ::decim], plf_c[:, ::decim]
-
-    psd /= n_epochs
-    plf = np.abs(plf) / n_epochs
+    psd = np.empty((n_channels, n_frequencies, n_times))
+    plf = np.empty((n_channels, n_frequencies, n_times), dtype=np.complex)
+    parallel, my_time_frequency, _ = parallel_func(_time_frequency, n_jobs)
+    psd_plf = parallel(my_time_frequency(data[:, c, :], Ws, use_fft)
+                       for c in range(n_channels))
+    for c, (psd_c, plf_c) in enumerate(psd_plf):
+        psd[c, :, :], plf[c, :, :] = psd_c[:, ::decim], plf_c[:, ::decim]
```

Comment:
It's relevant, with mkl I mostly run n jobs == 1

> On 10 Jan 2015, at 22:02, Eric Larson <notifications@github.com> wrote:
> 
> In mne/time_frequency/tfr.py:
> 
> > -                           for c in range(n_channels))
> > -
> > -        psd = np.zeros((n_channels, n_frequencies, n_times))
> > -        plf = np.zeros((n_channels, n_frequencies, n_times), dtype=np.complex)
> > -        for c, (psd_c, plf_c) in enumerate(psd_plf):
> > -            psd[c, :, :], plf[c, :, :] = psd_c[:, ::decim], plf_c[:, ::decim]
> > -
> > -    psd /= n_epochs
> > -    plf = np.abs(plf) / n_epochs
> > +    psd = np.empty((n_channels, n_frequencies, n_times))
> > +    plf = np.empty((n_channels, n_frequencies, n_times), dtype=np.complex)
> > +    parallel, my_time_frequency, _ = parallel_func(_time_frequency, n_jobs)
> > +    psd_plf = parallel(my_time_frequency(data[:, c, :], Ws, use_fft)
> > +                       for c in range(n_channels))
> > +    for c, (psd_c, plf_c) in enumerate(psd_plf):
> > +        psd[c, :, :], plf[c, :, :] = psd_c[:, ::decim], plf_c[:, ::decim]
> The memory consumption (and two copies) should be the same as it was for the n_jobs > 1 case, but you're right that it got worse for the n_jobs=1 case. I can bring the decim parameter into the parallel helper function to help with that a little bit. Is memory consumption a large concern here? It might be worth the simplification in code to have it be this way unless we see a case where we really need to trim consumption with n_jobs = 1 (which might be painfully slow).
> 
> —
> Reply to this email directly or view it on GitHub.
>